bus-mapping/src/circuit_input_builder/execution.rs

```diff
@@ -472,6 +472,8 @@ pub struct CopyEvent {
     pub copy_bytes: CopyBytes,
     /// Represents transaction access list
     pub access_list: Vec<CopyAccessList>,
+    /// Represents whether current execution bytecodes belong to first bytecode circuit.
+    pub is_first_bytecode_table: bool,
```

Comment:
it is a bit adhoc to add is_first_bytecode_table to `CopyEvent`.   src_id / dst_id (codehash acctually when copying bytecode?) can be used get is_first_bytecode_table?

Reply:
I am considering an idea: not changing the existing CopyEvent struct type. Instead,  
1. Modify the CopyCircuit struct to include a `&block` reference.  
  
2. Modify the copy circuit `assign_copy_events` helper function to include a block parameter.  
  
3. Modify the copy table `assignments` to also include a `&block` parameter, and then dynamically determine the 'is_first_bytecode' value there.

Reply:
sounds good

Reply:
this is done by importing `bytecode_map` to copy circuit assign, https://github.com/scroll-tech/zkevm-circuits/pull/1368/commits/5613f90679a0ea2ca00140da2db3c7cefa917265, CopyEvent keeps unchanged now.